pkg/client/restclient/request.go

```diff
@@ -105,16 +106,13 @@ type Request struct {
 	resource     string
 	resourceName string
 	subresource  string
-	selector     labels.Selector
 	timeout      time.Duration
 
 	// output
 	err  error
 	body io.Reader
 
-	// The constructed request and the response
-	req  *http.Request
-	resp *http.Response
+	ctx context.Context
```

Comment:
> That's just one example I can think of for needing timeouts and cancellation.

Ok, I'd really like to avoid using it like a magic container holding per-request things.  Can we at least doc that we don't plan to use to use to bypass a properly structured API for additional values?